pkg/scan/filter.go

```diff
@@ -16,74 +16,93 @@ const (
 var ValidAdvisoriesSets = []string{AdvisoriesSetResolved, AdvisoriesSetAll}
 
 // FilterWithAdvisories filters the findings in the result based on the advisories for the target APK.
-func FilterWithAdvisories(result *Result, advisoryCfgs *configs.Index[v2.Document], advisoryFilterSet string) ([]*Finding, error) {
+func FilterWithAdvisories(result *Result, advisoryDocIndices []*configs.Index[v2.Document], advisoryFilterSet string) ([]*Finding, error) {
 	if result == nil {
 		return nil, fmt.Errorf("result cannot be nil")
 	}
 
-	if advisoryCfgs == nil {
+	if advisoryDocIndices == nil {
 		return nil, fmt.Errorf("advisory configs cannot be nil")
 	}
 
-	documents := advisoryCfgs.Select().WhereName(result.TargetAPK.Origin()).Configurations()
+	var documents []v2.Document
+	for _, index := range advisoryDocIndices {
+		docs := index.Select().WhereName(result.TargetAPK.Origin()).Configurations()
+		documents = append(documents, docs...)
+	}
+
+	// TODO: Should we error out if we end up with multiple documents for a single package?
+
 	if len(documents) == 0 {
 		// No advisory configs for this package, so we know we wouldn't be able to filter anything.
 		return result.Findings, nil
 	}
 
-	// We know there's an advisories document for this package, so we can get the advisories.
-	packageAdvisories := documents[0].Advisories
+	// Use a copy of the findings so we don't mutate the original result.
```

Comment:
https://pkg.go.dev/golang.org/x/exp/slices#Clone

Reply:
Does that do the exact same thing?

Reply:
I think yes! Either one sgtm.